SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -360,8 +360,10 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
         
         // progressive decode the image in coder queue
         dispatch_async(self.coderQueue, ^{
+            // Get the finish status when progressive decoding task actually execute
+            BOOL finished = (self.receivedSize >= self.expectedSize);
             // If all the data has already been downloaded, earily return to avoid further decoding
-            if (self.receivedSize >= self.expectedSize) {
+            if (finished) {
```

Comment:
@zhongwuzw And, does this actually effect your use case ?
I introduce this feature, because some of user find that decoding big WebP may consume more time, so that even the download already finished, the progressive decoding is still happend until final task finished and show the final result.

This will only occur when your decoding code is slow (like use non-performant software codec on low CPU hardware). You can also check that [SDWebImageProgressiveJPEGDemo](https://github.com/SDWebImage/SDWebImageProgressiveJPEGDemo).

So, do we need to expect the finish status here ? Like change it to this:

```objective-c
// We only do final progressive decoding(with finished=YES) once. All the subsequence task dispatched to the coder queue between first entry and the last finished entry should be ignored.
if (finished && !downloadFinished) {
}
```